nemo/utils/get_rank.py

```diff
@@ -24,25 +24,10 @@ def is_global_rank_zero():
     if rank:
         return rank == 0
 
-    # If not set by pytorch, we need to determine node_rank
-    def get_node_rank():
-        # Use an equivalent of pytorch lightning's determine_ddp_node_rank()
-        node_rank = 0
-        # First check if running on a slurm cluster
-        # TODO: This check could probably be better
-        num_slurm_tasks = get_envint("SLURM_NTASKS", 0)
-        if num_slurm_tasks > 0:
-            node_rank = get_envint("SLURM_NODEID", 0)
-        else:
-            node_rank_env = get_envint("NODE_RANK", None)
-            group_rank = get_envint("GROUP_RANK", None)
-            if group_rank:
-                node_rank = group_rank
-            # Take from NODE_RANK whenever available
-            if node_rank_env:
-                node_rank = node_rank_env
-        return node_rank
-
-    node_rank = get_node_rank()
-    local_rank = get_envint("LOCAL_RANK", 0)
+    # on SLURM variables SLURM_NODEID and SLURM_PROCID will be defined
+    # SLURM_PROCID holds the global rank and computing the node_rank is
```

Comment:
As stated in this comment and looking at lightning's SLURM checks (https://github.com/PyTorchLightning/pytorch-lightning/blob/46f718d2ba31fdbb8f2abbef03471fec66204b66/pytorch_lightning/plugins/environments/slurm_environment.py)
"SLURM_PROCID" holds the global_rank, whereas "SLURM_LOCALID" holds the local rank. Shouldn't localrank be "SLURM_LOCALID"?
I guess at the end of the day, this function checks for global_rank==0 in which case the logic is correct but the variable names are not.

Reply:
Let's just change this section of code to be check if we are in a SLURM environment. If so just return get_envint("SLURM_PROCID") == 0 or in that line.

Reply:
OK.